snapshot/snapshot.go

```diff
@@ -137,9 +137,29 @@ func Verify(in io.Reader) (*raft.SnapshotMeta, error) {
 	if err := read(decomp, &metadata, ioutil.Discard); err != nil {
 		return nil, fmt.Errorf("failed to read snapshot file: %v", err)
 	}
+
+	if err := concludeGzipRead(decomp); err != nil {
+		return nil, err
+	}
```

Comment:
In general I agree with what you're saying, but I try to avoid leaving loaded footguns lying around if it's fairly cheap to do so, so prefer to just not hand back non-zero return values when an error occurs to prevent being misleading.